src/unix/tempdir.rs

```diff
@@ -144,6 +144,7 @@ mod tests {
         let path = t.as_path();
         assert!(path.exists());
         assert!(path.is_dir());
+        #[cfg(not(target_os = "macos"))]
```

Comment:
Can we instead check that the path is in the equivalent temp_dir for mac os? Otherwise this test is not very useful.